Document Deploy legacy property type migrators, import and migration flow and clarify grid editor configuration (#6533)
umbraco/UmbracoDocs: 13/umbraco-deploy/deployment-workflow/import-with-migrations.md

}
 ```

### Import and migration flow

When an import is started, the following happens:

> 1. Artifact signatures are read from the import provider (using `IArtifactImportProvider.GetArtifactSignatures()`).

Reviewer: Before I start working on this list, I need to know: Is it steps that the user needs to go through, or is it an explanation of what happens during a migration/import? 🤔

Author: This is an explanation of the things that happen and should give users a better understanding of the (order of) operations.

We also hadn't documented the notifications yet or the fact a default `ValidateArtifactImportDependenciesNotificationHandler` is added by Deploy. We might want to add a separate section about notifications though, as we also have notifications that happen during export (`ArtifactExportingNotification` and `ArtifactExportedNotification`).

Reviewer: Thanks. I wont touch the list then 💪 

It would be great to have a new section about notifications.
Let me know what you need from us to set this up!

#### Grid to Block Grid

> The Grid editor introduced in Umbraco 7 can still be used in version 13, but has been removed from Umbraco 14. It's functionality is replaced with the Block Grid editor.

Reviewer: ```suggestion
The Grid editor introduced in Umbraco 7 can still be used in version 13 but has been removed from Umbraco 14. Its functionality is replaced with the Block Grid editor.
```

}
```

{% hint style="info" %}

> The project you're importing into needs to know about any custom legacy Grid editor configurations to correctly migrate to the Block Grid editor. Make sure to either copy the `grid.editors.config.js` and `package.manifest` (containing grid editors) files or override the `GetGridEditors()` method of the artifact migrator to provide this.

Reviewer: ```suggestion
The project you are importing into needs to know about any custom legacy Grid editor configurations to migrate to the Block Grid editor correctly. Make sure to either copy the `grid.editors.config.js` and `package.manifest` (containing grid editors) files or override the `GetGridEditors()` method of the artifact migrator to provide this.
```

- `ReplaceDocTypeGridEditorDataTypeArtifactMigrator` extends `ReplaceGridDataTypeArtifactMigrator` and ensures any DocTypeGridEditor is migrated to blocks using the allowed element types. If the element types aren't found the default implementation will migrate to new element types.
- `DocTypeGridEditorPropertyTypeMigrator` extends `GridPropertyTypeMigrator` and ensures the Doc Type Grid Editor values are mapped one-to-one to the block item data.

{% hint style="info" %}

> The artifact migrator adds the default DTGE grid editor configuration (with alias `docType`), which can be disabled by setting the `AddDefaultDocTypeGridEditor` property to `false` (in a custom/inherited class). Similar to the base migrator, any custom DTGE grid editor configurations need to be available to correctly migrate to the Block Grid editor.

Reviewer: ```suggestion
The artifact migrator adds the default DocTypeGridEditor configuration (with alias `docType`). This can be disabled by setting the `AddDefaultDocTypeGridEditor` property to `false` in a custom/inherited class. Similar to the base migrator, any custom DocTypeGridEditor configurations must be available to migrate to the Block Grid editor correctly.
```
